solution_collector/src/main.rs

```diff
@@ -12,31 +14,104 @@ use github_rs::client::{Github, Executor};
 use std::collections::HashMap;
+use structopt::StructOpt;
 
-#[derive(Debug)]
-enum Mode {
+#[derive(StructOpt, Debug)]
+#[structopt(name = "solution_collector")]
+struct Opt {
+    #[structopt(subcommand)]
+    command: Command
+}
+
+#[derive(StructOpt, Debug)]
+enum Command {
+    #[structopt(name = "preview")]
     Preview,
-    Merge
+    #[structopt(name = "week-report")]
+    WeekReport {
+        week: u32
+    },
+    #[structopt(name = "merge")]
+    Merge {
+        week: u32,
+    }
+}
+
+struct Context {
+    owner: String,
+    repo: String,
+    client: Github
+}
+
+impl Context {
+    fn new(token: String, owner: String, repo: String) -> Self {
+        let client = Github::new(token).expect("Invalid token");
+        Context {
+            owner,
+            repo,
+            client
+        }
+    }
+}
+
+#[derive(Debug)]
+struct Problem {
+    title: String,
+    done: bool
+}
+
+#[derive(Debug)]
+struct SubmitEntry {
+    week: u32,
+    owner: String,
+    problems: Vec<Problem>
 }
 
-fn get_week(title: &str) -> u64 {
+fn get_week(title: &str) -> u32 {
     let re = Regex::new(r"(?i)week\s*(\d+)").unwrap();
     if let Some(cap) = re.captures(title) {
-        return cap.get(1).map_or(0, |w| w.as_str().parse::<u64>().unwrap_or(0));
+        return cap.get(1).map_or(0, |w| w.as_str().parse::<u32>().unwrap_or(0));
     } else {
         return 0;
     }
 }
 
-fn print_report(client: Github, owner: String, repo: String, w: u64) {
-    let mut summaries: HashMap<u64, String> = HashMap::new();
+fn parse_problems(body_str: &str) -> Vec<Problem> {
+    let re = Regex::new(r"(?i)\[.*\](.*)").unwrap();
+    let re_check_done = Regex::new(r"\[.*[^\s].*\]").unwrap();
+    body_str.lines().filter_map(|line| {
+        if let Some(cap) = re.captures(line) {
+            let title = cap.get(1).map_or(String::from(""), |m| m.as_str().to_string());
+            let done = if re_check_done.is_match(line) { true } else { false };
+            Some(Problem { title, done })
+        } else {
+            None
+        }
+    }).collect()
+}
+
+fn parse_pull_request(pr: Value) -> Option<SubmitEntry> {
+    let body_str = pr["body"].as_str().unwrap();
+    let re = Regex::new(r"(?i)Problem").unwrap();
+    if re.is_match(body_str) {
+        let week = get_week(pr["title"].as_str().unwrap());
+        let owner = pr["user"]["login"].as_str().unwrap().to_string();
+        let problems = parse_problems(body_str);
+        Some(SubmitEntry { week, owner, problems })
+    } else {
+        None
+    }
+}
+
+fn preview(ctx: Context) {
+    let mut summaries: HashMap<u32, String> = HashMap::new();
     let mut page = 1;
 
     loop {
-        let endpoint = format!("repos/{}/{}/pulls?page={}", owner, repo, page);
+        let endpoint = format!("repos/{}/{}/pulls?page={}", ctx.owner, ctx.repo, page);
         println!("Fetching Page {}", page);
 
-        let (_, _, pulls)  = client.get()
+        let (_, _, pulls)  = ctx.client.get()
```

Comment:
Could you filtered out all PR that are submitted after the deadline, or better mark them as due-date :D